src/Core/Services/StateService.cs

```diff
@@ -291,6 +291,20 @@ public async Task<bool> CanAccessPremiumAsync(string userId = null)
             return organizations?.Any(o => o.UsersGetPremium && o.Enabled) ?? false;
         }
 
+        public async Task SetPremiumAsync(bool value, string userId = null)
+        {
+            var reconciledOptions = ReconcileOptions(new StorageOptions { UserId = userId },
+                await GetDefaultStorageOptionsAsync());
+            var account = await GetAccountAsync(reconciledOptions);
+            if (account?.Profile == null || account.Profile.HasPremiumPersonally.GetValueOrDefault() == value)
```

Comment:
As we are using null conditional terminator when accessing `HasPremiumPersonally` on `CanAccessPremiumAsync`  added it here. I also believe it should not be necessary.

I'm checking if the value is different from the current one in order to avoid an unnecessary call to `SaveAccountAsync`